server/src/main/java/com/linecorp/centraldogma/server/internal/api/MirroringServiceV1.java

```diff
@@ -0,0 +1,152 @@
+     * POST /projects/{projectName}/mirrors
+     *
+     * <p>Creates a new mirror.
+     */
+    @Post("/projects/{projectName}/mirrors")
+    @ConsumesJson
+    @StatusCode(201)
+    public CompletableFuture<Revision> createMirror(@Param String projectName, MirrorDto newMirror,
+                                                    Author author) {
+        validateMirror(newMirror);
+        return mds.createMirror(projectName, newMirror, author);
+    }
+
+    private static void validateMirror(MirrorDto mirror) {
+        checkArgument(!Strings.isNullOrEmpty(mirror.id()), "Mirror ID is empty");
+        final Cron schedule = MirrorConfig.cronParser.parse(mirror.schedule());
+        final CronField secondField = schedule.retrieve(CronFieldName.SECOND);
+        checkArgument(!secondField.getExpression().asString().contains("*"),
+                      "The second field of the schedule must be specified. (seconds: *, expected: 0-59)");
+    }
+
+    /**
+     * GET /projects/{projectName}/mirrors/{index}
+     *
+     * <p>Returns the mirror at the specified index in the project mirror list.
+     */
+    @Get("/projects/{projectName}/mirrors/{index}")
+    public CompletableFuture<MirrorDto> getMirror(@Param String projectName, @Param int index) {
+        checkArgument(index >= 0, "index: %s (expected: >= 0)", index);
+
+        return projectManager().get(projectName).metaRepo().mirrors(true).thenApply(mirrors -> {
+            if (index >= mirrors.size()) {
+                throw new EntryNotFoundException(
+                        "No such mirror at the index " + index + " in " + projectName);
+            }
+            return convertToMirrorDto(projectName, mirrors.get(index));
+        });
+    }
+
+    /**
+     * PUT /projects/{projectName}/mirrors/{index}
+     *
+     * <p>Update the exising mirror.
+     */
+    @Put("/projects/{projectName}/mirrors/{index}")
+    @ConsumesJson
+    public CompletableFuture<Revision> updateMirror(@Param String projectName, @Param int index,
+                                                    MirrorDto mirror, Author author) {
+        validateMirror(mirror);
+        return mds.updateMirror(projectName, index, mirror, author);
+    }
+
+    private static MirrorDto convertToMirrorDto(String projectName, Mirror mirror) {
+        final URI remoteRepoUri = mirror.remoteRepoUri();
+        return new MirrorDto(mirror.index(),
+                             mirror.id(),
+                             projectName,
+                             mirror.schedule().asString(),
+                             mirror.direction().name(),
+                             mirror.localRepo().name(),
+                             mirror.localPath(),
+                             remoteRepoUri.getScheme(),
+                             remoteRepoUri.getHost() + remoteRepoUri.getPath(),
+                             mirror.remotePath(),
+                             firstNonNull(mirror.remoteBranch(), "master"),
```

Comment:
I'm not sure why we need to fallback to `master`.  If I remember correctly, @trustin added a capability to detect the default branch by default if this value is null, and `master` is no longer the de-facto if due to `main`.

Is this necessary for the UI?

Reply:
I guess we can leave it as `null` if a user specified `null` (= use default branch).